Rust: Support Integer enums using Serde_repr

bertkdowns opened the pull request:
<!-- Enter details of the change here. Include additional tests that have been done, reference to the issue for tracking, etc. -->

<!-- Please check the completed items below -->
### PR checklist
 
- [x ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [ x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request. @jacob-pro @paladinzh

This #9976 solved this issue but was never merged. This PR incorporated the same changes, and updates it to work with the current version of the app.

closes #9967, closes #18840 

An example from my app: The following couldn't be used, because it expected strings as explained in #18840 

![image](https://github.com/user-attachments/assets/bb05b5ca-9f68-44f2-9c02-996f1940cbdb)

Now, it generates to integer enums as expected:
![image](https://github.com/user-attachments/assets/b2178b7b-2eef-4ed5-9f63-30d5eeec038a)

First time contributor - LMK if there's anything else I've missed or need to do :)

wing328 commented on modules/openapi-generator/src/main/resources/rust/model.mustache:
```diff
@@ -6,11 +6,39 @@ use serde::{Deserialize, Serialize};
 {{^isEnum}}{{#vendorExtensions.x-rust-has-byte-array}}
 use serde_with::serde_as;
 {{/vendorExtensions.x-rust-has-byte-array}}{{/isEnum}}
+{{#isEnum}}{{#isInteger}}use serde_repr::{Serialize_repr,Deserialize_repr};{{/isInteger}}{{/isEnum}}
```
what about the following to avoid the empty line break?

```
{{#isEnum}}
{{#isInteger}}
use serde_repr::{Serialize_repr,Deserialize_repr};
{{/isInteger}}
{{/isEnum}}
```

bertkdowns commented:
Agreed, that's much cleaner. Wasnt sure how to remove that, i've updated, synced with master, and re-gen the apis, so hopefully all the workflows should pass now too.

wing328 commented on samples/server/petstore/kotlin-springboot-delegate/src/main/kotlin/org/openapitools/api/PetApi.kt:
```diff
@@ -59,7 +59,7 @@ interface PetApi {
     )
     fun addPet(@Parameter(description = "Pet object that needs to be added to the store", required = true) @Valid @RequestBody pet: Pet,serverHttpRequest: ServerHttpRequest): ResponseEntity<Pet> {
-        return getDelegate().addPet(pet, serverHttpRequest)
+        return getDelegate().addPet(pet, )
```
hmm... why there are changes to kotlin samples?

in the latest msater, the samples are up-to-date. can you please merge the latest master into your branch and regenerate the samples?

bertkdowns replied:
Hmm, that's strange, I already did that and merging with master is what caused the kotlin examples to change for me. Perhaps it's an environment thing (I'm using the devcontainers environment and `bin/generate-samples.sh`) , or i'll try rollback to an earlier commit where the changes don't happen.

wing328 commented:
can you please add a test spec for integer enum to `modules/openapi-generator/src/test/resources/3_0/rust/petstore.yaml` so that the CI can test it moving forward?

bertkdowns commented:
Ok, I think I've added those tests correctly - just involved updating the yaml file, so that the CI can check that it build correctly right? And dropped those changes to the kotlin files, must be an environment issue on my end

ogtn commented:
Thanks a lot for that PR !
